--- a/Marius.Pinta/src/core.c
+++ b/Marius.Pinta/src/core.c
@@ -133,6 +133,8 @@
     core->threads = thread;
 
     core->cache = pinta_core_cache_init(memory);
+    if (core->cache == NULL)
+        return NULL;
 
     return core;
 }
@@ -268,6 +270,8 @@
     PintaHeapCache *result;
 
     result = (PintaHeapCache*)pinta_memory_alloc(memory, sizeof(PintaHeapCache));
+    if (result == NULL)
+        return NULL;
 
     for (index = 0; index < 101; index++)
     {
@@ -407,6 +411,9 @@
         return &core->null_type;
 
     kind = object->block_kind;
+#ifdef PINTA_TEST_TRACE_TYPE
+    PINTA_TEST_TRACE_TYPE(object);
+#endif
     pinta_assert(kind < PINTA_KIND_LENGTH);
 
     return &core->types[kind];
